sound/soc/sof/ipc4-loader.c

```diff
@@ -203,11 +260,131 @@ static int sof_ipc4_query_fw_configuration(struct snd_sof_dev *sdev)
 out:
 	kfree(msg.data_ptr);
 
+	if (ret)
+		return ret;
+
+	/*
+	 * Unset the SOF_FW_PARSE_MANIFEST_PRE_BOOT flag to allow parsing the extended manifest in
+	 * the base firmware (id: 0)
+	 */
+	return sof_ipc4_fw_parse_ext_man(sdev, plat_data->fw, 0, 0);
+}
+
+static int sof_ipc4_load_library(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
+{
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	const guid_t *widget_uuid = &swidget->uuid;
+	const char *fw_filename;
+	size_t fw_offset;
+	int ret, i;
+
+	/* nothing to do if widget UUID is not set */
+	if (!guid_is_null(&swidget->uuid))
+		return 0;
+
+	/* check if the platform supports load_library */
+	if (sof_ops(sdev)->load_library) {
```

Comment:
Hrm, how this could work? The check is reversed, it should be `if (!sof_ops(sdev)->load_library)`